src/GraphQL.MicrosoftDI/GraphQLBuilder.cs

```diff
@@ -124,38 +124,59 @@ public IServiceRegister Register(Type serviceType, object implementationInstance
         }
 
         /// <inheritdoc/>
-        public IServiceRegister TryRegister(Type serviceType, Func<IServiceProvider, object> implementationFactory, ServiceLifetime serviceLifetime)
+        public IServiceRegister TryRegister(Type serviceType, Func<IServiceProvider, object> implementationFactory, ServiceLifetime serviceLifetime, RegistrationCompareMode mode = RegistrationCompareMode.ServiceType)
         {
             if (serviceType == null)
                 throw new ArgumentNullException(nameof(serviceType));
             if (implementationFactory == null)
                 throw new ArgumentNullException(nameof(implementationFactory));
 
-            ServiceCollection.TryAdd(new ServiceDescriptor(serviceType, implementationFactory, TranslateLifetime(serviceLifetime)));
+            var descriptor = new ServiceDescriptor(serviceType, implementationFactory, TranslateLifetime(serviceLifetime));
+            if (mode == RegistrationCompareMode.ServiceType)
+                ServiceCollection.TryAdd(descriptor);
+            else if (mode == RegistrationCompareMode.ServiceTypeAndImplementationType)
+                ServiceCollection.TryAddEnumerable(descriptor);
```

Comment:
We should add tests to hit all new lines of code.  I've got the rest of the Register/TryRegister methods at 99% coverage and it should not be hard to duplicate some of the tests to check TryAddEnumerable and the ArgumentOutOfRange exceptions.

Reply:
https://app.codecov.io/gh/graphql-dotnet/graphql-dotnet/blob/develop/src/GraphQL.MicrosoftDI/GraphQLBuilder.cs

Reply:
done